src/gazetteer-style.cpp

```diff
@@ -48,20 +61,26 @@ void gazetteer_style_t::clear()
     m_is_named = false;
 }
 
-bool gazetteer_style_t::has_place(std::string const &cls) const
+std::string gazetteer_style_t::class_list() const
 {
-    return std::any_of(m_main.begin(), m_main.end(), [&](pmaintag_t const &e) {
-        if (strcmp(std::get<0>(e), cls.c_str()) == 0) {
-            if (std::get<2>(e) & SF_MAIN_NAMED) {
-                return !m_names.empty();
-            }
-            // XXX should handle SF_MAIN_NAMED_KEY as well
+    std::string l;
 
-            return true;
+    for (auto const &m : m_main) {
+        bool do_include = true;
+        if (std::get<2>(m) & SF_MAIN_NAMED) {
```

Comment:
Why the `do_include`? What's up with the `XXX`?

Reply:
Because I thought the if would be too complicated once the XXX is solved. On further thinking, the XXX needs to be solved in a totally different place. Something for another PR. (Note that the XXX was there before, so no regression here.)